src/teleop/gui/src/components/EmulsionTest.vue

```diff
@@ -58,25 +63,42 @@ export default {
       serviceType: "mrover/ChangeServoAngle",
     });
 
-    let param = new ROSLIB.Param({
+    let servo_id_by_site = new ROSLIB.Param({
+      ros: this.$ros,
+      name: "science/servo_id_by_site",
+    });
+
+    let servo_positions = new ROSLIB.Param({
       ros: this.$ros,
-      name: "science/syringe_servo_positions",
+      name: "science/servo_positions",
     });
+
     //get servo angles from yaml file
-    param.get((angles) => {
-      let letter = "A";
-      for (var i = 0; i < 3; i++) {
-        this.angles[letter].pushed = angles["site_" + letter]["pushed"];
-        this.angles[letter].start = angles["site_" + letter]["start"];
-        letter = String.fromCharCode(letter.charCodeAt(0) + 1);
+    servo_id_by_site.get((servoIDsBySite) => {
+      for (let site in servoIDsBySite) {
+        let servoNum = servoIDsBySite[site];
```

Comment:
Does this make sense? We're iterating through servo_id_by_site, then iterating through that (somehow?), then taking the iterated value and using it to index into the same data structure. I could be misunderstanding it but I think this needs to be made more clear.

Reply:
Also would probably be helpful to not name the local variable servo_id_by_site since we already have that at the component level. I know I've said things like that are cool in the past but that's in more of a small constructor sort of setting, not a big complicated function like this